### PR TITLE
refactor $bindings to $inputs and $outputs

### PR DESCRIPTION
**Pull request by mariakleiner**
actually keeping backward compatibility turned out to be quite easy, so i decided to keep it, just in order to reduce the stress of breaking everything. if that's ok.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "arcsjs-core",
```
Comment (mariakleiner) [inline package-lock.json:8]: i messed up, i don't know why this change was made, but also in the spirit of my today's git adventure, i cannot find a way to revert it... i'll keep trying.

Comment (sjmiles) [reply package-lock.json:8]: You can ignore changes to package-lock.json, commit away

File: pkg/ts/core/Arc.ts [hunk 2]
```diff
@@ -78,7 +78,9 @@ export class Arc extends EventEmitter {
     this.log(`storeChanged: "${storeId}"`);
     values(this.hosts).forEach((host: Host) => {
       const bindings = host.meta?.bindings;
-      const isBound = bindings && entries(bindings).some(([n, v]) => (v || n) === storeId);
+      const outputs = host.meta?.outputs;
+      const isBound = (bindings && entries(bindings).some(([n, v]) => (v || n) === storeId))
```
Comment (sjmiles) [inline pkg/ts/core/Arc.ts:82]: factor out: `(<list> && entries(<list>).some(([n, v]) => (v || n) === storeId))`

File: pkg/ts/core/Arc.ts [hunk 3]
```diff
@@ -95,14 +97,20 @@ export class Arc extends EventEmitter {
   protected computeInputs(host) {
     const inputs = nob();
     const bindings = host.meta?.bindings;
+    const recipeInputs = host.meta?.inputs;
```
Comment (sjmiles) [inline pkg/ts/core/Arc.ts:100]: I believe these are "inputBindings"?

File: pkg/ts/recipe/ParticleCook.ts [hunk 4]
```diff
@@ -30,8 +30,17 @@ export class ParticleCook {
   }
   static specToMeta(spec) {
     // TODO(sjmiles): impedance mismatch here is likely to cause problems
-    const {$kind: kind, $container: container, $staticInputs: staticInputs, $bindings: bindings} = spec;
-    return {kind, staticInputs, bindings, container};
+    const {$kind: kind, $container: container, $staticInputs: staticInputs, $bindings: bindings, $inputs: inputs, $outputs: outputs} = spec;
```
Comment (sjmiles) [inline pkg/ts/recipe/ParticleCook.ts:33]: Should we add $ to the StoreMeta keywords?

**Review by sjmiles — APPROVED** [review]

**Pull request closed without merging**